enterprise_access/apps/api_client/lms_client.py

```diff
@@ -170,6 +193,22 @@ def fetch_group_members(
         }
         if is_reversed:
             params['is_reversed'] = is_reversed
+        if traverse_pagination:
+            cache_key = all_enterprise_group_members_cache_key(
+                group_uuid,
+                sort_by,
+                user_query,
+                show_removed,
+                is_reversed,
+            )
+            cached_response = TieredCache.get_cached_response(cache_key)
+            if cached_response.is_found:
+                logger.info(
+                    f'all_enterprise_group_members cache hit for group_uuid {group_uuid}.'
+                )
+                return cached_response.value
+
+            params['page_size'] = 500
```

Comment:
Why are we setting page_size to 500?

Reply:
When traversing all pages of group member data, whether we are gathering CSV data or sorting by enrollment count, we want to make as few requests to platform as possible. When testing, I've found that some of our backends support page sizes up to 100, and a few others- 500. Django will just cap at whatever limit it has set if you request over that so I figured just make it super large to ensure we make as few requests as possible.